cli/CommandExecutor.cpp

```diff
@@ -0,0 +1,200 @@
+         rel_it != catalog_database->end(); ++rel_it) {
+      int column_width = static_cast<int>(rel_it->getName().length());
+      max_column_width =
+          max_column_width < column_width ? column_width : max_column_width;
+      relation_count++;
+    }
+  } else {
+    const ParseString &table_name = arguments->front();
+    const std::string &table_name_val = table_name.value();
+    relation = catalog_database->getRelationByName(table_name_val);
+    int column_width = static_cast<int>(relation->getName().length());
+    max_column_width =
+        max_column_width < column_width ? column_width : max_column_width;
+    relation_count++;
+  }
+  // only if we have relations
+  if (relation_count > 0) {
+    vector<int> column_widths;
+    column_widths.reserve(relation_count);
+    column_widths.push_back(max_column_width);
+    column_widths.push_back(max_column_width);
+    fputs("       List of relations\n\n", out);
+    fprintf(out, "%-*s|", max_column_width, " Name");
+    fprintf(out, "%-*s|", 6, " Type");
+    fputc('\n', out);
+    PrintToScreen::printHBar(column_widths, out);
+    //  if there are no argument print the entire list of tables
+    //  else print the particular table only
+    if (arguments->size() == 0) {
+      for (auto rel_it = catalog_database->begin();
+           rel_it != catalog_database->end(); ++rel_it) {
+        fprintf(out, " %-*s|", max_column_width, rel_it->getName().c_str());
+        fprintf(out, " %-*s", max_column_width, "table");
+        fputc('\n', out);
+      }
+    } else {
+      fprintf(out, " %-*s|", max_column_width, relation->getName().c_str());
+      fprintf(out, " %-*s", max_column_width, "table");
+      fputc('\n', out);
+    }
+    fputc('\n', out);
+  }
+}
+
+void CommandExecutor::executeDescribeTable(
+    const PtrVector<ParseString> *arguments,
+    const CatalogDatabase *catalog_database, FILE *out) {
+  const ParseString &table_name = arguments->front();
+  const std::string &table_name_val = table_name.value();
+  const CatalogRelation *relation =
+      catalog_database->getRelationByName(table_name_val);
+  vector<int> column_widths;
+  // initialize it to the number of columns that need to be printed
+  column_widths.reserve(2);
+
+  int max_column_width = 0;
+  for (CatalogRelation::const_iterator attr_it = relation->begin();
+       attr_it != relation->end(); ++attr_it) {
+    // Printed column needs to be wide enough to print:
+    //   1. The attribute name (in the printed "header").
+    //   2. Any value of the attribute's Type.
+    //   3. If the attribute's Type is nullable, the 4-character string "NULL".
+    // We pick the largest of these 3 widths as the column width.
+    int column_width = static_cast<int>(attr_it->getDisplayName().length());
+    column_width = column_width < attr_it->getType().getPrintWidth()
+                       ? attr_it->getType().getPrintWidth()
+                       : column_width;
```

Comment:
Include `<algorithm>`, and refactor to `column_width = std::max(column_width, attr_it->getType().getPrintWidth())`, so that we don't call the virtual function twice.